packages/contracts/contracts/registry/Registry.sol

```diff
@@ -4,36 +4,49 @@
 
 pragma solidity 0.8.10;
 
-import "../core/DAO.sol";
+import "../core/component/Component.sol";
+
+/// @title Register a DAO
+/// @author Aragon Association - 2022
+/// @notice This contract allows to register a DAO by its address
+contract Registry is Component {
```

Comment:
- Are we adding the Component base features to get upgradability and permissions?
- Is there any extra feature included that the Registry doesn't need?
- Will we have to grant Aragon the REGISTER_DAO_ROLE?

Reply:
> Are we adding the Component base features to get upgradability and permissions?
Yes.
> Is there any extra feature included that the Registry doesn't need?
We also inherit [ERC165](https://eips.ethereum.org/EIPS/eip-165), which allows the registry to publish its interface and detect if another contract supports an interface, which is useful to make the registry future-proof).

We also thought about making it a `MetaTxComponent` (so that someone else could pay for using it) but as this one is will be used mostly through the `DAOFactory` (which is a `MetaTxComponent`) and everything happens within one transaction, it is sufficient that `DAOFactory` supports meta-txs if we want to have this feature.

> Will we have to grant Aragon the REGISTER_DAO_ROLE?
@Rekard0 and I decided that it makes sense to deploy an "AdminDAO" for the entire registration process (also for the `DAOFactory`). 
The AdminDAO will manage all the parts. In the beginning, it could be controlled by a single wallet or multi-sig. Later it can become it's own DAO or a child-/sub-DAO of the AN DAO. We have full flexibility.